cou/steps/plan.py

```diff
@@ -265,11 +269,67 @@ def _verify_data_plane_ready_to_upgrade(args: CLIargs, analysis_result: Analysis
     """
     if args.upgrade_group in {DATA_PLANE, HYPERVISORS}:
         if not analysis_result.min_o7k_version_data_plane:
-            raise DataPlaneCannotUpgrade(
-                "Cannot find data-plane apps. Is this a valid OpenStack cloud?"
+            PlanStatus.add_message(
+                "Cannot find data-plane apps. Is this a valid OpenStack cloud?",
+                MessageType.ERROR,
             )
         if not _is_control_plane_upgraded(analysis_result):
-            raise DataPlaneCannotUpgrade("Please, upgrade control-plane before data-plane")
+            PlanStatus.add_message(
+                "Please upgrade control-plane before data-plane",
+                MessageType.ERROR,
+            )
+
+
+async def _verify_vault_is_unsealed(analysis_result: Analysis) -> None:
+    """Verify if vault is unsealed.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :raises DataPlaneCannotUpgrade: When data-plane is not ready to upgrade.
+    """
+    try:
+        await verify_vault_is_unsealed(analysis_result.model)
+    except VaultSealed as e:
+        PlanStatus.add_message(str(e), MessageType.ERROR)
+
+
+async def _verify_osd_noout_unset(analysis_result: Analysis) -> None:
+    """Verify Ceph OSD has noout unset.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :raises DataPlaneCannotUpgrade: When data-plane is not ready to upgrade.
+    """
+    try:
+        await ceph.assert_osd_noout_state(
+            analysis_result.model, analysis_result.apps_control_plane, state=False
+        )
+    except ApplicationError:
+        PlanStatus.add_message(
+            "'noout' should be set for ceph cluster",
```

Comment:
This message contradicts the function definition - should `noout` be set or unset?